### PR TITLE
End user/fix/home search

### PR DESCRIPTION
**Pull request by ay0o5h**

### CODE REVIEW

**Review by ahmednasserzaza — APPROVED** [review]

**Review by aya-seif — APPROVED** [review]

**Review by devsadeq — APPROVED** [review]

File: client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt [hunk 1]
```diff
@@ -72,14 +76,15 @@ class HomeScreen : BaseScreen<
...
+
         when (effect) {
             is HomeScreenUiEffect.NavigateToCuisineDetails -> println("Cuisine id ${effect.cuisineId}")
             is HomeScreenUiEffect.NavigateToCuisines -> navigator.root?.push(CuisinesScreen())
             is HomeScreenUiEffect.NavigateToChatSupport -> navigator.root?.push(ChatSupportScreen())
             is HomeScreenUiEffect.NavigateToOrderTaxi -> println("Navigate to Order Taxi screen")
             is HomeScreenUiEffect.ScrollDownToRecommendedRestaurants -> println("Scroll down home screen")
             is HomeScreenUiEffect.NavigateToOfferItem -> println("Navigate to offer item details ${effect.offerId}")
-            is HomeScreenUiEffect.NavigateToSearch -> println("Navigate to Search Screen")
+            is HomeScreenUiEffect.NavigateToSearch -> navigator to SearchTab
```
Comment (mosayed01) [inline client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt:87]: I guess we don't need an effect to handle tab navigation see:

File: client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt [hunk 2]
```diff
@@ -133,11 +139,14 @@ class HomeScreen : BaseScreen<
...
-                    onValueChange = {},
-                    onClick = { listener.onClickSearch() },
+                    onValueChange = listener::onChangeSearchText,
+                    onClick = {
+                        listener.onClickSearch()
+                        tabNavigator.current = SearchTab
+                              },
                     leadingPainter = painterResource(Resources.images.searchOutlined),
                     modifier = Modifier.padding(horizontal = 16.dp)
                 )
```
Comment (mosayed01) [inline client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt:152]:
```
                BpSimpleTextField(
                    text = state.searchTerm,
                    hint = Resources.strings.searchHint,
                    hintColor = Theme.colors.contentSecondary,
                    enabled = false, // we need to pass enabled to text field with value false
                    onValueChange = listener::onChangeSearchText,
                    leadingPainter = painterResource(Resources.images.searchOutlined),
                    modifier = Modifier.padding(horizontal = 16.dp)
                        .noRippleEffect { tabNavigator.current = SearchTab } // use noRipple modifier directly to the hole parent
                )
```

File: design_system/shared/src/commonMain/kotlin/ui/composable/BpSimpleTextField.kt [hunk 3]
```diff
@@ -52,7 +52,6 @@ fun BpSimpleTextField(
                     hint,
                     style = Theme.typography.body,
                     color = hintColor,
-                    modifier = Modifier.noRippleEffect(onClick)
                 )
             },
             onValueChange = onValueChange,
```
Comment (mosayed01) [inline design_system/shared/src/commonMain/kotlin/ui/composable/BpSimpleTextField.kt:57]:
```suggestion
            enabled = enabled,
            onValueChange = onValueChange,
```

File: client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt (hunk 2 above)
Comment (ay0o5h) [reply client_end_user/shared/src/commonMain/kotlin/presentation/home/HomeScreen.kt:152]: i need to click icon and it use noRippleEffect in texst already  the modifier is for all text not just icon i don't think i need to add on click on whole text

File: design_system/shared/src/commonMain/kotlin/ui/composable/BpSimpleTextField.kt (hunk 3 above)
Comment (ay0o5h) [reply design_system/shared/src/commonMain/kotlin/ui/composable/BpSimpleTextField.kt:57]: for what shoul i add enabled ?

**Review by mosayed01 — APPROVED** [review]

**Pull request merged**